tests/integration/test_row_checks.py

```diff
@@ -1044,3 +1046,178 @@ def test_col_is_valid_timestamp(spark, set_utc_timezone):
+            ["Value '192.168..1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1.' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '.192.168.1.1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192. 168.1.1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1. 1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1.1 ' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value ' 192.168.1.1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.01.1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '001.002.003.004' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '256.168.1.1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '1.2.3.256' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '-1.0.0.0' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '0.0.0.-1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '1.1.1.1000' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1.0/24' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1.0/0' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1.0/32' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1.0/33' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1.0/abc' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192.168.1.0/' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '/24' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '12345' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '19..2.168.1.1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '192....168.1.1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+            ["Value '1.1.1.1.1.1.1.1' in Column 'a' does not match pattern 'IPV4_ADDRESS'"],
+        ],
+        checked_schema,
+    )
+    assert_df_equality(actual, expected, ignore_nullable=True)
+
+
+def test_is_ipv4_in_cidr(spark):
+    schema_ipv4 = "a: string, b: string"
+
+    test_df = spark.createDataFrame(
+        [
+            ["255.255.255.255", "192.168.01.1"],
+            ["0.0.0.0", "192.168.1"],
+            ["abc.def.ghi.jkl", None],
+            ["255255155255", "127.0.0.1"],
+            ["192.168.1.1", "10.0.0.1"],
+            ["172.16.0.1", "224.0.0.1"],
+            ["240.0.0.1", "192.168.1"],
+            ["192.168.1.1.1", ""],
+            [" ", "abc.def.ghi.jkl"],
+            ["1.178.7.255", "1.178.4.0"],
```

Comment:
`1.178.4.0` Column 2, and column 2 `1.178.4.255` on line 1175 & 1176 are positive checks. Do you want me to add more to it? @mwojtyczka